### PR TITLE
Crud user

### PR DESCRIPTION
**Pull request by Iryna97**
Please review and confirm.
@kukuruzavictor I merged UserServiceIMPL and UserController could you please take a look if your logic was not changed.

### CODE REVIEW

**Review by kukuruzavictor — APPROVED** [review]

File: src/test/java/com/feedback/service/EmailSenderServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@SpringBootTest
+public class EmailSenderServiceTest {
+
+    @Test
+    public void testSendEmail(){
```
Comment (Forfend) [inline src/test/java/com/feedback/service/EmailSenderServiceTest.java:15]: Please, format the code

File: src/test/java/com/feedback/service/AuthenticationServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        UserDto userDto = authenticationService.register(map(user()));
+        assertNotNull(userDto);
+        assertEquals(map(user()), userDto);
+        verify(userRepo, times(1)).findUserByEmail("test@mail.com");
+        verify(userRepo, times(1)).saveAndFlush(map(dto,user().getPassword()));
+        verify(emailSenderService, times(1)).sendEmail(email());
+    }
+
+    @Test(expected = UserAlreadyExistException.class)
+    public void testRegisterExistingUser() throws UserAlreadyExistException{
```
Comment (Forfend) [inline src/test/java/com/feedback/service/AuthenticationServiceImplTest.java:72]: Please, format the code

File: src/main/java/com/feedback/service/UserServiceImpl.java [hunk 3]
```diff
@@ -30,35 +27,37 @@
 import static com.feedback.dto.FeedbackDto.map;
 
 @Service
-@Transactional
-public class UserServiceImpl implements UserService {
+public class UserServiceImpl implements UserService{
```
Comment (Forfend) [inline src/main/java/com/feedback/service/UserServiceImpl.java:30]: Please, format the code

File: src/main/java/com/feedback/service/AuthenticationServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    private static final String SECRET_WORD = "id-secret";
+
+    public AuthenticationServiceImpl(UserRepo userRepo, EmailSenderService emailSenderService, PasswordEncoder passwordEncoder) {
+        this.userRepo = userRepo;
+        this.emailSenderService = emailSenderService;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    @Override
+    public UserDto register(UserDto userDto){
```
Comment (Forfend) [inline src/main/java/com/feedback/service/AuthenticationServiceImpl.java:31]: Please, format the code

**Review by Forfend — CHANGES_REQUESTED** [review]
Please, format the code using shot-ket ```CRTL + ALT + L```

**Review by Forfend — APPROVED** [review]

**Comment by Forfend** [comment]
image
compilation error

**Review by Forfend — CHANGES_REQUESTED** [review]
fix compilation error

**Pull request merged**